pkg/lang/ir/python.go

```diff
@@ -151,19 +151,28 @@ func (g Graph) compilePyPIPackages(root llb.State) llb.State {
 	if g.RequirementsFile != nil {
 		// Compose the package install command.
 		var sb strings.Builder
-		sb.WriteString("/opt/conda/envs/envd/bin/python -m pip install -r ")
-		sb.WriteString(*g.RequirementsFile)
+		sb.WriteString("bash -c '")
+		sb.WriteString("set -euo pipefail\n")
+		sb.WriteString(fmt.Sprintf("chown -R envd:envd %s\n", g.getWorkingDir())) // Change mount dir permission
+		envdCmd := strings.Builder{}
+		envdCmd.WriteString(fmt.Sprintf("cd %s\n", g.getWorkingDir()))
+		envdCmd.WriteString(fmt.Sprintf("/opt/conda/envs/envd/bin/python -m pip install -r  %s\n", *g.RequirementsFile))
+
+		// Execute the command to write yaml file and conda env using envd user
+		sb.WriteString(fmt.Sprintf("sudo -i -u envd bash << EOF\n%s\nEOF\n", envdCmd.String()))
+		sb.WriteString("'")
 		cmd := sb.String()
+
 		logrus.WithField("command", cmd).
 			Debug("Configure pip install requirements statements")
-		root = root.Dir(g.getWorkingDir())
+		root = root.User("root").Dir(g.getWorkingDir())
 		run := root.
 			Run(llb.Shlex(cmd), llb.WithCustomNamef("pip install %s",
 				strings.Join(g.PyPIPackages, " ")))
 		run.AddMount(cacheDir, cache,
 			llb.AsPersistentCacheDir(g.CacheID(cacheDir), llb.CacheMountShared), llb.SourcePath("/cache"))
 		run.AddMount(g.getWorkingDir(),
-			llb.Local(flag.FlagBuildContext), llb.Readonly)
```

Comment:
no.it's one directionn change won't be synced to host

Reply:
Then could we keep llb.Readonly here?

Reply:
No. Basically because `-e .` installation needs to create some temp files under the current directory. And mounted directory is owned by root. So here I need to run the command under user root, and use envd user to run some codes in the middle

Reply:
LGTM.